wcfsetup/install/files/lib/system/session/SessionHandler.class.php

```diff
@@ -425,97 +436,91 @@ public function getUser() {
 	}
 	
 	/**
-	 * Tries to read existing session identified by the given session id.
-	 * 
-	 * @param	string		$sessionID
+	 * Tries to read existing session identified by the given session id. Returns whether
+	 * a session could be found.
 	 */
-	protected function getExistingSession($sessionID) {
-		$this->session = new $this->sessionClassName($sessionID);
-		if (!$this->session->sessionID) {
-			$this->session = null;
-			return;
+	protected function getExistingSession(string $sessionID): bool {
+		$sql = "SELECT	*
+			FROM	wcf".WCF_N."_".($this->isACP ? 'acp' : 'user')."_session
+			WHERE	sessionID = ?";
+		$statement = WCF::getDB()->prepareStatement($sql);
+		$statement->execute([
+			$sessionID
+		]);
+		$row = $statement->fetchSingleRow();
+		
+		if (!$row) {
+			return false;
 		}
 		
-		$this->user = new User($this->session->userID);
-		if ($this->isACP) {
-			$this->virtualSession = ACPSessionVirtual::getExistingSession($sessionID);
+		// Check whether the session technically already expired.
+		$lifetime =
+			($this->isACP   ? self::ACP_SESSION_LIFETIME  :
+			($row['userID'] ? self::USER_SESSION_LIFETIME :
+			(                 self::GUEST_SESSION_LIFETIME)));
```

Comment:
I think an explicit, lengthier piece of code would make it easier to comprehend than nested ternary operators.

Reply:
see: https://github.com/WoltLab/WCF/pull/3593#discussion_r503777640

Reply:
```php
if ($this->isACP) {
    $lifetime = self::ACP_SESSION_LIFETIME;
}
else if ($row['userID']) {
    $lifetime = self::USER_SESSION_LIFETIME;
}
else {
    $lifetime = self::GUEST_SESSION_LIFETIME;
}
```

I do not see why this formatting would have any downside.

Reply:
- It's more than twice as long, causing less of the method to fit on screen.
- It repeats the `$lifetime =` variable for each case, putting more words on the screen.
- It requires diagonal scans to check the condition against the value, whereas the ternary is formatted in a nice columnar laylout.